Index: guest_arm_toIR.c
===================================================================
--- guest_arm_toIR.c	(revision 2817)
+++ guest_arm_toIR.c	(working copy)
@@ -17247,7 +17247,7 @@
                   unop(Iop_32to1,
                        binop(Iop_Xor32,
                              mkexpr(r15guard), mkU32(1))),
-                  r15kind,
+                  r15kind == Ijk_Ret ? Ijk_Boring : r15kind,
                   IRConst_U32(guest_R15_curr_instr_notENC + 4),
                   OFFB_R15T
          ));
