diff -rupw VEX-pyvex/priv/guest_ppc_toIR.c VEX-fish/priv/guest_ppc_toIR.c
--- VEX-pyvex/priv/guest_ppc_toIR.c	2014-02-25 04:46:17.925740902 -0800
+++ VEX-fish/priv/guest_ppc_toIR.c	2014-02-25 04:47:24.773742585 -0800
@@ -5658,6 +5658,7 @@ static Bool dis_branch ( UInt theInstr,
             vex_printf("dis_int_branch(ppc)(bcctr,BO)\n");
             return False;
          }
+
          DIP("bcctr%s 0x%x, 0x%x\n", flag_LK ? "l" : "", BO, BI);
          
          assign( cond_ok, branch_cond_ok( BO, BI ) );
@@ -5669,6 +5670,7 @@ static Bool dis_branch ( UInt theInstr,
          if (flag_LK)
             putGST( PPC_GST_LR, e_nia );
          
+        if(!((BO & 0x14) == 0x14))
          stmt( IRStmt_Exit(
                   binop(Iop_CmpEQ32, mkexpr(cond_ok), mkU32(0)),
                   Ijk_Boring,
@@ -5709,6 +5711,7 @@ static Bool dis_branch ( UInt theInstr,
          if (flag_LK)
             putGST( PPC_GST_LR,  e_nia );
 
+         if (!vanilla_return)
          stmt( IRStmt_Exit(
                   binop(Iop_CmpEQ32, mkexpr(do_branch), mkU32(0)),
                   Ijk_Boring,
